### PR TITLE
Add more options for ajaxUpload

### PR DESCRIPTION
**Pull request by dardub**
I think the following upload options should be available for complete flexibility:
- http methods (POST, PATCH, DELETE)
- file input name
- setting header (needed for many authorization methods)

### CODE REVIEW

**Comment by dardub** [comment]
I'm thinking now I should have also included the dist builds.

**Comment by PowerKiKi** [comment]
Would you please add documentation and tests ?

**Comment by dardub** [comment]
Sure. I'll try and find some time tomorrow. Did you want me to also commit the dist builds?

**Comment by lightswitchr** [comment]
Yes please! Merge, merge, merge!

**Comment by PowerKiKi** [comment]
@lightswitchr if you contribute documentation and tests, it will get merged

File: src/uploader.js [hunk 1]
```diff
@@ -126,6 +129,12 @@ function uiUploader($log) {
         // Append file data:
         formData.append(key, file, file.name);
 
+        if (headers.length) {
```
Comment (lightswitchr) [inline src/uploader.js]: This won't work as objects don't have a length.

The following code worked for me:

if (headers) {
            for (var header_key in headers) {
                if (headers.hasOwnProperty(header_key)) {
                    xhr.setRequestHeader(header_key, headers[header_key]);
                }
            }
        }

**Comment by lightswitchr** [comment]
@PowerKiKi It's alright asking for tests, but you have no existing tests for the project anyway.Asserting true == true is not a test...

**Comment by dardub** [comment]
Sorry. I've been on vacation and playing catch up on projects. I haven't forgotten about this.

**Comment by steambao** [comment]
Awesome, setting headers for things like BasicAuth much needed. +1 on a1c66ca3d077326382e00cc3b1a787661d00eb6f

**Comment by PowerKiKi** [comment]
FYI, headers support were added in #19

**Comment by PowerKiKi** [comment]
This PR is unlikely to be merged as is now. But if somebody still need it and rebase it as a new PR, it could get merged.

**Pull request closed without merging**